### PR TITLE
Training test migration

### LINKED ISSUE #907: [FEATURE]: Migrate Pytests from /backend to /training
### Feature Name

Migrate Pytests from /backend to /training

### Your Name

Daniel Wu

### Description

Write the following tests for /training in /training/tests:

- test_imports.py: Tests if all imports used by the training backend work
- test_loss_function.py: Tests if the loss functions in criterion.py (somewhere in /training/core) work
- test_model_parser.py: Tests if DlModel can parse LayerParams into a model
- test_model.py: Tests if DlModel can correctly build a model
- test_sk_learn_default_dataset_train_test.py: Tests if SklearnDatasetCreator can correctly split a default dataset into train and test datasets given the dataset's name

Use pytest testing framework to implement tests

### PR DESCRIPTION
**Pull request by prishavall**
What user problem are we solving?
Migrating unit tests from backend to the training module.

What solution does this PR provide?
Fixes the pathway errors and import changes with the transfer from backend to training.
Testing Methodology
How did you test your changes and verify that existing
functionality is not broken
Ran tests individually
Any other considerations
One test model_py is still not working need to look into django implementation for that.

### CODE REVIEW

**Comment by karkir0003** [comment]
@prishavall I got the build check to work for this action. Had to do some fixing with the micromamba action

File: training/tests/test_loss_function.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    [
+        (zero_tensor, one_tensor, 1.0),
+        (tensor_vstack_one, tensor_vstack_two, 15543340),
+    ],
+)
+def test_mse_loss_computation_correct(output, labels, expected_number):
+    loss_function_name = "MSELOSS"
+    loss_function = getCriterionHandler("MSELOSS")
+    computed_loss = compute_loss(loss_function_name, output, labels)
+    print(torch.sqrt(computed_loss))
```
Comment (karkir0003) [inline training/tests/test_loss_function.py]: remove print

File: training/tests/test_model.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    [
+        ([nn.Linear(10, 5), nn.Linear(5, 3)]),
+        ([nn.Linear(0, 0), nn.Linear(0, 0)]),
+        ([nn.Linear(100, 50), nn.Linear(5, 3)]),
+    ],
+)
+def test_dlmodel(input_list):
+    print("input_list: " + str(input_list) + " is of type " + str(type(input_list)))
+    my_model = DLModel(input_list)
+    print("my_model: " + str(my_model) + " is of type " + str(type(my_model)))
```
Comment (karkir0003) [inline training/tests/test_model.py]: remove this print

File: training/tests/test_model.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    print(
```
Comment (karkir0003) [inline training/tests/test_model.py]: remove this print

File: training/training/core/dl_model.py [hunk 4]
```diff
@@ -41,6 +39,7 @@ def fromLayerParamsList(cls, layer_params_list: list[LayerParams]):
         return cls(layer_list)
 
     def build_model(self, layer_list):
+        print("Hello World")
```
Comment (karkir0003) [inline training/training/core/dl_model.py]: remove this print

**Review by karkir0003 — CHANGES_REQUESTED** [review]
remove debug prints (nits)

File: training/tests/test_loss_function.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+@pytest.mark.parametrize(
+    "output, labels, expected_number",
+    [
+        (zero_tensor, one_tensor, 1.0),
+        (tensor_vstack_one, tensor_vstack_two, 15543340),
+    ],
+)
+def test_mse_loss_computation_correct(output, labels, expected_number):
+    loss_function_name = "MSELOSS"
+    loss_function = getCriterionHandler("MSELOSS")
```
Comment (karkir0003) [inline training/tests/test_loss_function.py]: this line doesn't seeem to be used. You can remove it

**Review by karkir0003 — CHANGES_REQUESTED** [review]
added another quick comment

**Review by karkir0003 — APPROVED** [review]
